src/async-components/views/dialogs/security/NewRecoveryMethodDialog.tsx

```diff
@@ -66,7 +65,7 @@ export default class NewRecoveryMethodDialog extends React.PureComponent<IProps>
         const hackWarning = <p className="warning">{_t("encryption|new_recovery_method_detected|warning")}</p>;
 
         let content: JSX.Element | undefined;
-        if (MatrixClientPeg.safeGet().getKeyBackupEnabled()) {
+        if (this.props.newVersionInfo) {
```

Comment:
`this.props.newVersionInfo` is of type `IKeyBackupInfo`. Won't that always be truthy?

Reply:
Right, this code was a bit confusing and I added a bit more.

So I looked at what it was doing, and updated it to make a bit more sense.
So the idea is that when we detect that the backup is removed/updated we want to notify the user.
I don't think it was that usefull to make a difference if the new backup is trusted or not as anyhow it was offering the option to go to settings. And from settings you can do what you want.

So basically there are 2 cases:
- There is no backup anymore: Shows the dialog to setup a new one as a shortcut or go to settings
- There is a new backup: Shows the dialog that notifies you with an option to open the settings.

I added playwrigth tests to test it for rust crypto.
Unfortunatly the detection of change on legacy crypto is harder to trigger.. I disabled the test for legacy for now.

I think that if we really want this feature, we should do it better and improve the code to detect backup changes

Reply:
The other option would be to remove that feature until we have clearer requirements